### PR TITLE
Add config for enable FlexibleHttpClient and HttpMethodsClient

### PR DESCRIPTION
**Pull request by Nyholm**
This will address part of #70 

With config options you wrap your client in FlexibleHttpClient or HttpMethodsClient. Please give input on the config names. I want them to be descriptive but at the same time also reflect the class we use for wrapping. 

A descriptive config name would be "emulate_async_client" but that is very far from the name `FlexibleHttpClient`. I guess you understand my point. Any suggestions are welcome.

### CODE REVIEW

File: DependencyInjection/HttplugExtension.php [hunk 1]
```diff
@@ -206,4 +196,55 @@ private function configureAuthentication(ContainerBuilder $container, array $con
...
+                ->addArgument($arguments['config']);
+        }
+
+
+        /*
+         * Decorate the client with clients from client-common
+         */
+
+        if ($arguments['flexible_client'] && $arguments['add_http_methods']) {
+            throw new \LogicException('A http client can\'t be decorated with both FlexibleHttpClient and HttpMethodsClient. Only one of the following options can be true. ("flexible_client", "add_http_methods")');
```
Comment (dbu) [inline DependencyInjection/HttplugExtension.php]: this should be validated in the Configuration class. you can add a `validate()` section (or was it `validation()`?) on the level that has the two options.

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -102,6 +102,14 @@ protected function configureClients(ArrayNodeDefinition $root)
                         ->cannotBeEmpty()
                         ->info('The service id of a factory to use when creating the adapter.')
                     ->end()
+                    ->booleanNode('flexible_client')
+                        ->defaultFalse()
+                        ->info('If true we will wrap this client in a FlexibleHttpClient which emulates async or sync behavior.')
```
Comment (dbu) [inline DependencyInjection/Configuration.php]: i would say: 

Set to true to get the client wrapped in a FlexibleHttpClient which emulates...

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -102,6 +102,14 @@ protected function configureClients(ArrayNodeDefinition $root)
...
+                    ->end()
+                    ->booleanNode('add_http_methods')
```
Comment (dbu) [inline DependencyInjection/Configuration.php]: i think calling this http_methods_client would be fine

**Comment by dbu** [comment]
the name flexible_client seems right to me. i would call the other http_methods_client.

**Comment by Nyholm** [comment]
Thank you for your review. I've update the PR according to your suggestions

**Comment by dbu** [comment]
thanks a lot @Nyholm !
created https://github.com/php-http/documentation/issues/107 for the documentation.

**Comment by Nyholm** [comment]
Thank you for merging.

**Pull request merged**